src/apps/answers/tests/test_answers.py

```diff
@@ -4693,3 +4619,278 @@ async def test_include_ehr_parameter_in_data_endpoint(
         )
 
         assert _answer["ehrDataFile"] == filename
+
+
+@pytest.mark.usefixtures("mock_kiq_report")
+class TestCompletions(BaseTest):
+    fixtures = [
+        "workspaces/fixtures/workspaces.json",
+    ]
+    applet_answers_completions_url = "/answers/applet/{applet_id}/completions"
+    applets_answers_completions_url = "/answers/applet/completions"
+
+    @pytest.mark.asyncio
+    async def test_empty_applets_version_map(
+        self,
+        client: TestClient,
+        global_session: AsyncSession,
+    ):
+        # Create a new user with no applet assignments
+        user_crud = UsersCRUD(global_session)
+        new_user = await _get_or_create_user(
+            user_crud,
+            UserCreate(
+                email=f"empty-{uuid.uuid4()}@example.com",
+                password="Test1234!",
+                first_name="Empty",
+                last_name="User",
+            ),
+            global_session,
+            uuid.uuid4(),
+        )
+        await global_session.commit()
+        await global_session.close()
+
+        client.login(new_user)
+
+        # --- Case 1: filterByVersion=true ---
```

Comment:
For cases like this one, pytest has something called parameterized tests:

```python
@pytest.mark.parametrize("filter_by_version", [True, False])
async def test_applets_completions_with_filter_option(self, client, tom, answer, filter_by_version):
  # ...
```

Using this can reduce code duplication